### PR TITLE
Extract from url

### PR DESCRIPTION
**Pull request by nitzspiz**

### CODE REVIEW

File: src/pydomainextractor.cpp [hunk 1]
```diff
@@ -504,3 +557,7 @@ PyInit_pydomainextractor(void) {
 
     return m;
 }
+
+int main(){
```
Comment (nitzspiz) [inline src/pydomainextractor.cpp:561]: ignore, it's a late addition in order to compile the CPP independently

File: src/pydomainextractor.cpp [hunk 2]
```diff
@@ -421,6 +499,12 @@ static PyMethodDef DomainExtractor_methods[] = {
         METH_FASTCALL,
         "Extract a domain string into its parts\n\nextract(domain)\nArguments:\n\tdomain(str): the domain string to extract\nReturn:\n\ttuple(str, str, str): subdomain, domain, suffix\n\n"
     },
+    {
+        "extract_from_url",
+        (PyCFunction)DomainExtractor_extract_from_url,
+        METH_FASTCALL,
+        "Checks whether a domain is a valid domain\n\nextract_from_url(domain)\nArguments:\n\tdomain(str): the domain string to validate\nReturn:\n\tbool: True if valid or False if invalid\n\n"
```
Comment (wavenator) [inline src/pydomainextractor.cpp:506]: fix the function documentation please

File: tests/test_pydomainextractor.py [hunk 3]
```diff
@@ -472,3 +472,569 @@ def test_load_called_with_data(
...
+        self,
+    ):
+        result = self.domain_extractor.extract_from_url(
+            'http://admin:password1@www.google.com:666/secret/admin/interface?param1=42'
+        )
+        assert result == {
+            'subdomain': 'www',
+            'domain': 'google',
+            'suffix': 'com',
+        }
```
Comment (wavenator) [inline tests/test_pydomainextractor.py:1040]: please put all these tests in their appropriate function names. Take domain test and put them inside the test case of the domains, leave URLs in one function

File: setup.py [hunk 4]
```diff
@@ -5,7 +5,7 @@
 
 setuptools.setup(
     name='PyDomainExtractor',
-    version='0.6.1',
+    version='0.6.2',
```
Comment (wavenator) [inline setup.py:8]: bump it to 0.7.0 as there is an API change

File: setup.py [hunk 5]
```diff
@@ -44,6 +44,7 @@
             ],
             extra_link_args=[
                 '-lidn2',
+                '-lcpprest',
```
Comment (wavenator) [inline setup.py:47]: make sure to add this dependency inside the github CI/Cd file. Also add it to the readme.md

File: setup.py (hunk 4 above)
Comment (wavenator) [reply setup.py:8]: Sorry, I've just checked and we are currently on 0.7.0. Bump it to 0.8.0

File: tests/test_pydomainextractor.py (hunk 3 above)
Comment (nitzspiz) [reply tests/test_pydomainextractor.py:1040]: I didn't get you, all url extraction functions are under TLDExtractTestCase.
Do you mean you want to see one function under the ExtractionTestCase, that contains all the url tests?
`
class DomainExtractorExtractionTestCase:
    def test_extract_from_url():
        # test with protocol
        # test with protocol and query fragments
       ....
`

**Pull request closed without merging**